### PR TITLE
mark stargz configuration as experimental

### PR DESCRIPTION
**Pull request by changweige**
As the stargz compatiblity feature is not production ready
 and reliable enought

Signed-off-by: Changwei Ge <gechangwei@bytedance.com>

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -175,6 +178,7 @@ type SnapshotterConfig struct {
 	ImageConfig        ImageConfig        `toml:"image"`
 	CacheManagerConfig CacheManagerConfig `toml:"cache_manager"`
 	LoggingConfig      LoggingConfig      `toml:"log"`
+	Experimental       Experimental       `toml:"experimental"`
```
Comment (imeoer) [inline config/config.go:181]: Will it move out once the feature becomes ready?

Comment (changweige) [reply config/config.go:181]: Yes. We will move it out when it's ready. In fact, the stargz compibility feature has malfunctioned for quite a time.

Comment (imeoer) [reply config/config.go:181]: How do we ensure the CLI flags compatibility when moving it out?

Comment (changweige) [reply config/config.go:181]: CLI does not has --enable-stargz flag now. and stargz feature has malfunctioned for a long time with stable nydusd release. So I suppose we don't have to worry about compatiblity.

File: config/config.go [hunk 2]
```diff
@@ -93,6 +93,10 @@ const (
 	FsDriverFscache string = "fscache"
 )
 
+type Experimental struct {
```
Comment (sctb512) [inline config/config.go:96]: Could we add an example in `toml` and related document?

Comment (changweige) [reply config/config.go:96]: Good point, already changed the example toml configuraion file

**Review by sctb512 — APPROVED** [review]
LGTM

**Comment by imeoer** [comment]
Need to be rebased.

**Comment by changweige** [comment]
> Need to be rebased.

Rebased. Thanks.

**Pull request merged**